modules/core/ante/ante.go

```diff
@@ -30,10 +32,19 @@ func (rrd RedundantRelayDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 		for _, m := range tx.GetMsgs() {
 			switch msg := m.(type) {
 			case *channeltypes.MsgRecvPacket:
-				response, err := rrd.k.RecvPacket(ctx, msg)
+				var (
+					response *channeltypes.MsgRecvPacketResponse
+					err      error
+				)
+				if ctx.IsCheckTx() {
```

Comment:
So confusing to have isCheckTx/isReCheckTx and `ExecMode()` 😅 

Just noting for reviewers that here we are in `CheckTxMode` explicitly because we've checked also got `!simulate` above on L28. 

On the v0.50+ line there is union between `IsCheckTx` and `ExecMode == execModeSimulate` (after https://github.com/cosmos/cosmos-sdk/pull/20346). i.e. `IsCheckTx` is true when in `execModeSimulate` (this is because simulateTx runs on the checkState multistore)

Reply:
I guess this is where it should be the other way around like you mentioned @colin-axner. When recheckTx=true then checkTx=true... because ¯\_(ツ)_/¯